Content.Server/Power/EntitySystems/BatterySystem.cs

```diff
@@ -100,6 +111,11 @@ private void OnEmpPulse(EntityUid uid, BatteryComponent component, ref EmpPulseE
         {
             args.Affected = true;
             UseCharge(uid, args.EnergyConsumption, component);
+            // Apply a cooldown to the entity's self recharge if needed to avoid it immediately self recharging after an EMP.
+            if (TryComp<BatterySelfRechargerComponent>(uid, out var batterySelfRechargerComponent))
+                if (batterySelfRechargerComponent.AutoRechargePause && batterySelfRechargerComponent.AutoRechargePauseTime > 0)
+                    if (args.EnergyConsumption > 0 && batterySelfRechargerComponent.AutoRechargePauseTime > batterySelfRechargerComponent.AutoRechargeCooldown)
+                        SetChargeCooldown(uid, batterySelfRechargerComponent.AutoRechargePauseTime, batterySelfRechargerComponent);
```

Comment:
This shouldn't be tightly coupled into other systems.

Reply:
Clarify?

Reply:
Cleaned it up.